### PR TITLE
feat: Add error validation for update all

### LINKED ISSUE #593: Copa might need to skip validation or find a different source for validation, as copa won't have the scanner input to validate against.

### PR DESCRIPTION
**Pull request by ashnamehrotra**
merge after #570 

Describe the changes in this pull request using active verbs such as _Add_, _Remove_, _Replace_ ...

Closes #593 

Part of #522

### CODE REVIEW

**Comment by ashnamehrotra** [comment]
tested this with hardcoding, added followup issue (#660) to add end to end testing as discussed

File: pkg/pkgmgr/apk.go [hunk 1]
```diff
@@ -200,8 +200,13 @@ func (am *apkManager) upgradePackages(ctx context.Context, updates unversioned.U
...
 	} else {
 		// if updates is not specified, update all packages
-		installCmd := `apk upgrade --no-cache`
-		apkInstalled = apkUpdated.Run(llb.Shlex(installCmd), llb.WithProxy(utils.GetProxy())).Root()
+		installCmd := `output=$(apk upgrade --no-cache 2>&1); if [ $? -ne 0 ]; then echo "$output" >>error_log.txt; fi`
+		apkInstalled = apkUpdated.Run(llb.Args([]string{"sh", "-c", installCmd}), llb.WithProxy(utils.GetProxy())).Root()
+
+		// Validate no errors were encountered if updating all
+		if !ignoreErrors {
+			apkInstalled = apkInstalled.Run(llb.Args([]string{"sh", "-c", "if [ -s error_log.txt ]; then cat error_log.txt; exit 1; fi"})).Root()
```
Comment (sozercan) [inline pkg/pkgmgr/apk.go:208]: can we create an util function for "sh" "-c"?

```
func Sh(cmd string) llb.RunOption {
	return llb.Args([]string{"/bin/sh", "-c", cmd})
}
```

File: pkg/pkgmgr/rpm.go [hunk 2]
```diff
@@ -489,10 +494,23 @@ func (rm *rpmManager) unpackAndMergeUpdates(ctx context.Context, updates unversi
...
+		done
+		`
 	}
 
-	downloaded := busyboxCopied.Run(llb.Shlex(downloadCmd), llb.WithProxy(utils.GetProxy())).Root()
+	downloaded := busyboxCopied.Run(llb.Args([]string{"bash", "-c", downloadCmd}), llb.WithProxy(utils.GetProxy())).Root()
+
+	// Validate no errors were encountered if updating all
+	if updates == nil && !ignoreErrors {
+		downloaded = downloaded.Run(llb.Args([]string{"bash", "-c", "if [ -s error_log.txt ]; then cat error_log.txt; exit 1; fi"})).Root()
```
Comment (sozercan) [inline pkg/pkgmgr/rpm.go:512]: do we need bash here or should we keep it consistent with sh with other places?

File: pkg/pkgmgr/rpm.go [hunk 3]
```diff
@@ -489,10 +494,23 @@ func (rm *rpmManager) unpackAndMergeUpdates(ctx context.Context, updates unversi
...
+			output=$(yumdownloader --downloadonly --downloaddir=. --best -y "$package" 2>&1)
+			if [ $? -ne 0 ]; then
+				echo "$output" >>error_log.txt
+			fi
+		done
+		`
 	}
 
-	downloaded := busyboxCopied.Run(llb.Shlex(downloadCmd), llb.WithProxy(utils.GetProxy())).Root()
+	downloaded := busyboxCopied.Run(llb.Args([]string{"bash", "-c", downloadCmd}), llb.WithProxy(utils.GetProxy())).Root()
```
Comment (sozercan) [inline pkg/pkgmgr/rpm.go:508]: same here

File: pkg/pkgmgr/apk.go (hunk 1 above)
Comment (sozercan) [reply pkg/pkgmgr/apk.go:208]: applies to other places we are calling "sh -c" too

File: pkg/pkgmgr/rpm.go (hunk 2 above)
Comment (ashnamehrotra) [reply pkg/pkgmgr/rpm.go:512]: we can use sh here

**Review by sozercan — APPROVED** [review]
Thank you! LGTM

**Pull request merged**